src/translate.py

```diff
@@ -1335,15 +1336,21 @@ def format(self, fmt: Formatter, force_dec: bool = False) -> str:
 
         prefix = ""
         suffix = ""
-        if not fmt.skip_casts:
+        if not fmt.skip_casts and not self.elide_cast:
             if self.type.is_pointer():
                 prefix = f"({self.type.format(fmt)})"
             if self.type.is_unsigned():
                 suffix = "U"
+
         if force_dec:
             value = str(self.value)
         else:
-            value = fmt.format_int(self.value)
+            value = fmt.format_int(
+                self.value,
+                str(self.type),
+                self.type.is_signed(),
+                self.type.get_size_bits(),
```

Comment:
I think you can use just `self.type.is_signed()` & `self.type.get_size_bits()` -- I don't think you also need to check `str(self.type)`? 

I think it would also be better to move the negative number logic into this function. I don't think `format_int()` needs to change at all?

Reply:
I thought it would have to go into format_int, since the change is specifically about formatting ints. Also seemed like I'd just be duplicating code otherwise.

Reply:
I think the role of `format_int` (and `format_hex`) is to convert a bare Python number to a string. `Literal.format()`s is converting a typed C literal: it's responsible for looking at the type information and deciding how to use it. For example, adding the `U` suffix is done in `Literal.format` and not as an argument to `format_int`.
I think doing it in `Literal.format()` will also keep the size of the change much smaller: `format_int()` won't need to change at all, and you won't need to pass a handful of boolean arguments around.

As for s8/s16/s32/s64, you can check `type.is_int() and type.is_signed()` instead of doing string comparison.